```
--- FAIL: TestNodeAddresses (0.02s)
    azure_instances_test.go:372: Test [NodeAddresses should get both ipV4 and ipV6 private addresses] unexpected error: invalid character 'v' after top-level value
    azure_instances_test.go:377: Test [NodeAddresses should get both ipV4 and ipV6 private addresses] unexpected ipAddresses: [], expected [{"Hostname" "vm1"} {"InternalIP" "10.240.0.1"} {"InternalIP" "1111:11111:00:00:1111:1111:000:111"}]
    azure_instances_test.go:392: Test [NodeAddresses should get both ipV4 and ipV6 private addresses] unexpected error: invalid character 'v' after top-level value
    azure_instances_test.go:396: Test [NodeAddresses should get both ipV4 and ipV6 private addresses] unexpected ipAddresses: [], expected [{"Hostname" "vm1"} {"InternalIP" "10.240.0.1"} {"InternalIP" "1111:11111:00:00:1111:1111:000:111"}]
    azure_instances_test.go:372: Test [NodeAddresses should get ipV4 private and public addresses] unexpected error: invalid character 'v' after top-level value
```

--- a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances_test.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances_test.go
@@ -350,7 +350,7 @@ func TestNodeAddresses(t *testing.T) {
 
 		mux := http.NewServeMux()
 		mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			fmt.Fprint(w, metadataTemplate, test.nodeName, test.ipV4, test.ipV4Public, test.ipV6, test.ipV6Public)
+			fmt.Fprintf(w, metadataTemplate, test.nodeName, test.ipV4, test.ipV4Public, test.ipV6, test.ipV6Public)
 		}))
 		go func() {
 			http.Serve(listener, mux)

--- a/staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances_test.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/azure/azure_instances_test.go
@@ -131,7 +131,7 @@ func TestInstanceID(t *testing.T) {
 
 		mux := http.NewServeMux()
 		mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			fmt.Fprint(w, fmt.Sprintf(`{"compute":{"name":"%s","subscriptionId":"subscription","resourceGroupName":"rg"}}`, test.metadataName))
+			fmt.Fprintf(w, "{\"compute\":{\"name\":\"%s\",\"subscriptionId\":\"subscription\",\"resourceGroupName\":\"rg\"}}", test.metadataName)
 		}))
 		go func() {
 			http.Serve(listener, mux)
@@ -350,7 +350,7 @@ func TestNodeAddresses(t *testing.T) {
 
 		mux := http.NewServeMux()
 		mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			fmt.Fprint(w, fmt.Sprintf(metadataTemplate, test.nodeName, test.ipV4, test.ipV4Public, test.ipV6, test.ipV6Public))
+			fmt.Fprint(w, metadataTemplate, test.nodeName, test.ipV4, test.ipV4Public, test.ipV6, test.ipV6Public)
 		}))
 		go func() {
 			http.Serve(listener, mux)

